drt-run: Fix warnings to avoid build failure

nameisbhaskar opened the pull request:
The build is failing on my mac due to the warnings. This PR fixes the same.
```
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
compilepkg: nogo: errors found by nogo during build-time code analysis:
pkg/cmd/drt-run/event.go:125:39: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/event.go:128:8: Unlock is >5 lines away from matching Lock, move it to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/event.go:131:16: unchecked error (errcheck)
pkg/cmd/drt-run/event.go:143:52: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/event.go:146:7: Unlock is >5 lines away from matching Lock, move it to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/event.go:149:16: unchecked error (errcheck)
pkg/cmd/drt-run/http.go:103:16: unchecked error (errcheck)
pkg/cmd/drt-run/main.go:94:20: unchecked error (errcheck)
pkg/cmd/drt-run/operations.go:112:29: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:119:25: function call between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:121:53: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:128:31: function call between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:140:4: for loop between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:158:8: Unlock is >5 lines away from matching Lock, move it to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:218:11: unchecked error (errcheck)
pkg/cmd/drt-run/operations.go:259:74: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:262:2: function call between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:263:7: Unlock is >5 lines away from matching Lock, move it to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/workloads.go:106:11: unchecked error (errcheck)
Target //pkg/cmd/drt-run:drt-run failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 85.126s, Critical Path: 61.50s
INFO: 1080 processes: 5 internal, 1075 darwin-sandbox.
ERROR: Build did NOT complete successfully
INFO: Build Event Protocol files produced successfully.
ERROR: exit status 1
```

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/131399)
<!-- Reviewable:end -->

srosenberg commented:
> The build is failing on my mac due to the warnings. 

Those are build errors, not warnings. Is this not failing in CI?

itsbilal commented on pkg/cmd/drt-run/operations.go:
```diff
@@ -108,54 +108,53 @@ func (r *opsRunner) pickOperation(
-
-		r.mu.Unlock()
+		}()
 		return opSpec
```
Some of the cases where were `continue`-ing in the loop earlier are no longer logically equivalent; we'll be returning from the closure and then returning the opSpec right here even if it's ineligible to be run. You might wanna add a return value to the function.

nameisbhaskar replied:
I initially added a boolean and then realised that, this has the entire code, so, it was same. But as I can understand from what you are saying, if someone adds more code post this function, the continue will not work as expected. Will add a boolean return.

itsbilal reviewed (commented):
Looks good save for one case of logical-nonequivalence.

nameisbhaskar commented:
Thanks @itsbilal for reviewing the changes. I have addressed the comments. PTAL.

vidit-bhat commented:
I think we can avoid `_, _ = fmt.Fprintf` changes since these are warnings and not errors

vidit-bhat commented on pkg/cmd/drt-run/event.go:
```diff
@@ -64,25 +64,25 @@ type Event struct {
 func (e *Event) String() string {
 	var builder strings.Builder
-	fmt.Fprintf(&builder, "[%s] ", e.Timestamp.Format(time.RFC3339))
+	_, _ = fmt.Fprintf(&builder, "[%s] ", e.Timestamp.Format(time.RFC3339))
```
I see `fmt.Fprintf` being used in multiple places in the codebase. I think we can remove these changes.

vidit-bhat commented on pkg/cmd/drt-run/event.go:
```diff
@@ -119,16 +119,18 @@ func (l *eventLogger) logWorkloadEvent(ev Event, workloadIdx int) {
+	}()
 	we.events[idx] = ev
-	io.WriteString(l.outputFile, ev.String()+"\n")
+	_, _ = io.WriteString(l.outputFile, ev.String()+"\n")
```
How about
```
_, err := io.WriteString(l.outputFile, ev.String()+"\n")
if err != nil {
	return // or print something useful
}
```

vidit-bhat commented on pkg/cmd/drt-run/http.go:
```diff
@@ -51,86 +51,86 @@ func (h *httpHandler) writeRunningOperations(rw http.ResponseWriter) {
-	fmt.Fprintf(rw, "<body>\n")
+	_, _ = fmt.Fprintf(rw, "<!DOCTYPE html>\n")
+	_, _ = fmt.Fprintf(rw, "<html lang=\"en\">\n")
+	_, _ = fmt.Fprintf(rw, "<head><title>DRT Run</title><meta charset=\"utf-8\"></head>\n")
+	_, _ = fmt.Fprintf(rw, "<body>\n")
 
-	fmt.Fprintf(rw, "<h3>Workloads</h3><hr>\n")
+	_, _ = fmt.Fprintf(rw, "<h3>Workloads</h3><hr>\n")
 
-	fmt.Fprintf(rw, "<p>Workloads:</p>\n<ul>")
+	_, _ = fmt.Fprintf(rw, "<p>Workloads:</p>\n<ul>")
```
```
fmt.Fprintf(rw, `<!DOCTYPE html>
<html lang="en">
<head><title>DRT Run</title><meta charset="utf-8"></head>
<body>
<h3>Workloads</h3><hr>
<p>Workloads:</p><ul>`)
```
Let's combine continuous Fprintf statements

vidit-bhat commented on pkg/cmd/drt-run/main.go:
```diff
@@ -91,7 +91,7 @@ func runDRT(configFile string) (retErr error) {
 		eventL: eventL,
 	}
-	hh.startHTTPServer(8080, "localhost")
+	_ = hh.startHTTPServer(8080, "localhost")
```
Better to handle the error
```
err = hh.startHTTPServer(8080, "localhost")
if err != nil {
	return err // or print something useful
}
```

vidit-bhat commented on pkg/cmd/drt-run/operations.go:
```diff
@@ -215,7 +218,7 @@ func (r *opsRunner) runOperation(
 		return
 	}
-	cmd.Start()
+	_ = cmd.Start()
```
```
if err := cmd.Start(); err != nil {
  return errors.Wrap(err, "start")
}
```

vidit-bhat commented on pkg/cmd/drt-run/http.go:
```diff
@@ -51,86 +51,86 @@ func (h *httpHandler) writeRunningOperations(rw http.ResponseWriter) {
 	encoder := yaml.NewEncoder(rw)
-	encoder.Encode(h.w.config)
-	fmt.Fprintf(rw, "</pre>")
+	_ = encoder.Encode(h.w.config)
```
```
if err := encoder.Encode(h.w.config); err != nil {
	panic(err)
}
```

nameisbhaskar commented:
> > The build is failing on my mac due to the warnings.
> 
> Those are build errors, not warnings. Is this not failing in CI?

It is not clear how it passed CI.

nameisbhaskar committed:
drt-run: Fix warnings to avoid build failure

The build is failing on my mac due to the warnings. This PR fixes the same.
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
compilepkg: nogo: errors found by nogo during build-time code analysis:
pkg/cmd/drt-run/event.go:125:39: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/event.go:128:8: Unlock is >5 lines away from matching Lock, move it to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/event.go:131:16: unchecked error (errcheck)
pkg/cmd/drt-run/event.go:143:52: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/event.go:146:7: Unlock is >5 lines away from matching Lock, move it to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/event.go:149:16: unchecked error (errcheck)
pkg/cmd/drt-run/http.go:103:16: unchecked error (errcheck)
pkg/cmd/drt-run/main.go:94:20: unchecked error (errcheck)
pkg/cmd/drt-run/operations.go:112:29: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:119:25: function call between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:121:53: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:128:31: function call between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:140:4: for loop between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:158:8: Unlock is >5 lines away from matching Lock, move it to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:218:11: unchecked error (errcheck)
pkg/cmd/drt-run/operations.go:259:74: if statement between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:262:2: function call between Lock and Unlock may be unsafe, move Unlock to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/operations.go:263:7: Unlock is >5 lines away from matching Lock, move it to a defer statement after Lock (deferunlockcheck)
pkg/cmd/drt-run/workloads.go:106:11: unchecked error (errcheck)
Target //pkg/cmd/drt-run:drt-run failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 85.126s, Critical Path: 61.50s
INFO: 1080 processes: 5 internal, 1075 darwin-sandbox.
ERROR: Build did NOT complete successfully
INFO: Build Event Protocol files produced successfully.
ERROR: exit status 1

Epic: none

Release note: None

nameisbhaskar commented:
Thanks @vidit-bhat for reviewing the changes. I have reverted all Fprintf changes and made only the necessary changes to make the build success. So, for the error handling and returning of errors, I will leave the way it is to keep the behaviour same and keep the change only to remove warnings for the build to succeed.

itsbilal reviewed (approved):
LGTM

nameisbhaskar commented:
Thanks @itsbilal and @vidit-bhat for reviewing the changes.

nameisbhaskar commented:
bors r=@itsbilal
